File: src/aula9/division_by_two_and_permutation/main.py
#https://codeforces.com/blog/entry/101380
import math

ntests = int(input())

for _ in range(ntests):
    n = int(input())

    numbers = list(map(lambda x : int(x), input().split(" ")))

    numbers.sort(reverse=True)

    used = [False] * (n+1)

    ok = True

    for no in numbers:
        x = no

        while (x > n or used[x]):
            x = math.floor(x/2)
        
        if (x > 0):
            used[x] = True
        else:
            ok = False
            break
    
    result = 'YES' if ok else 'NO'

    print(result)

File: src/aula9/division_by_two_and_permutation/main.cpp
#https://codeforces.com/blog/entry/101380

#include <bits/stdc++.h>

using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(0);
    int t;
    scanf("%d", &t);
//    cin >> t;
    while (t--) {
        int n;
        scanf("%d", &n);
        vector<int> nums(n), used(n + 1);
        for (auto &i : nums) {
            scanf("%d", &i);
        }
        sort(nums.begin(), nums.end(), [](int a, int b) { return a > b; });
        bool isOk = true;
        for (auto &num : nums) {
            int x = num;
            while (x > n || used[x]) {
                x /= 2;
            }
            if (x) used[x] = 1;
            else {
                isOk = false;
                break;
            }
        }
        printf("%s\n", isOk ? "YES" : "NO");
    }
}